crates/primitives/starknet/tests/block.rs

```diff
@@ -0,0 +1,80 @@
+
+
+#[cfg(test)]
+mod tests {
+    use mp_starknet::block::*;
+    use sp_core::{H256, U256};
+    use sp_core::Encode;
+    use mp_starknet::execution::ContractAddressWrapper;
+
+    fn generate_dummy_header() -> Header {
+        Header::new(
+            H256::from_slice(&[1; 32]),
+            U256::from(1),
+            U256::from(2),
+            ContractAddressWrapper::default(),
+            42,
+            0,
+            H256::from_slice(&[3; 32]),
+            0,
+            H256::from_slice(&[4; 32]),
+            Some(1),
+            Some(U256::from(3)),
+        )
+    }
+
+    #[test]
+    fn test_header_creation() {
```

Comment:
You are testing the `new` function there. Not really useful a long as it does not contain any custom logic.
If you want you can use `derive_more::Constructor` instead, so you just have to trust the lib and not test it